src/main/java/edu/harvard/iq/dataverse/engine/command/impl/UpdateDataverseCommand.java

```diff
@@ -90,20 +90,19 @@ public Dataverse execute(CommandContext ctxt) throws CommandException {
         
     @Override
     public boolean onSuccess(CommandContext ctxt, Object r) {
-        Dataverse result = (Dataverse) r;
-
-        List<Dataset> datasets = ctxt.datasets().findByOwnerId(result.getId());
+        
+        // first kick of async index of datasets
+        // TODO: is this actually needed? Is there a better way to handle
         try {
-            Future<String> indResponse = ctxt.index().indexDataverse(result);
+            Dataverse result = (Dataverse) r;
+            List<Dataset> datasets = ctxt.datasets().findByOwnerId(result.getId());
             ctxt.index().asyncIndexDatasetList(datasets, true);
```

Comment:
reindexing datasets may not actually be needed here. (based on the instances where it is called ManageTemplates, Guestbook, etc.)

Reply:
I think you're right, but I wanted to change as little as possible while surgically fixing the deadlocks. So left this code as it was, but added the TODO.